### PR TITLE
feat(switch): customisable icons for switch, controlled mode

### PR DESCRIPTION
**Pull request by Powerplex**
**TASK**: #618 

### Description, Motivation and Context

Following specs review session, it seems some features needed to be added:
- icon removal
- icon customization
- controlled mode demo

As well as two others that we need to discuss before merging:

### Types of changes
- [x] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🧾 Documentation
- [x] 📷 Demo
- [x] 🧠 Refactor
- [x] 💄 Styles

### Screenshots - Animations
Capture d’écran 2023-04-18 à 18 57 26

### CODE REVIEW

File: packages/hooks/use-combined-state/src/useCombinedState.tsx
Comment (andresin87) [inline packages/hooks/use-combined-state/src/useCombinedState.tsx:1]: ✅

**Review by kikoruiz — APPROVED** [review]

File: packages/components/switch/src/SwitchInput.styles.ts [hunk 1]
```diff
@@ -60,20 +58,25 @@ export const thumbStyles = cva(
...
     },
-    defaultVariants,
   }
 )
+
+export const thumbCheckSVGStyles = cva(['transition-opacity duration-200'], {
+  variants: {
+    size: makeVariants<'size'>({
+      sm: ['h-sz-10 w-sz-10'],
+      md: ['h-sz-12 w-sz-12'],
```
Comment (andresz1) [inline packages/components/switch/src/SwitchInput.styles.ts:76]: Should the classes be separated by a comma ? you are doing it above

Comment (andresin87) [inline packages/components/switch/src/SwitchInput.styles.ts:76]:
```suggestion
      sm: ['h-sz-10, 'w-sz-10'],
      md: ['h-sz-12, 'w-sz-12'],
```

Comment (andresin87) [reply packages/components/switch/src/SwitchInput.styles.ts:76]:
```suggestion
      sm: ['h-sz-10', 'w-sz-10'],
      md: ['h-sz-12', 'w-sz-12'],
```

**Pull request merged**